test/instrumentation/httpdc/utils.test.ts

```diff
@@ -0,0 +1,829 @@
+          name.toLowerCase() === 'authorization' ? authHeader : undefined,
+        socket: {
+          remotePort: port,
+        } as Socket,
+      };
+    }
+
+    it('should return absolute url with localhost', () => {
+      const req = mockRequest({ path: '/test/1' });
+      const result = utils.getAbsoluteUrl(req as ClientRequest);
+      assert.strictEqual(result, 'http://localhost/test/1');
+    });
+
+    it('should return absolute url', () => {
+      const req = mockRequest({
+        path: '/test/1?query=1',
+        host: 'www.google',
+      });
+      const result = utils.getAbsoluteUrl(req as ClientRequest);
+      assert.strictEqual(result, 'http://www.google/test/1?query=1');
+    });
+
+    it('should return default url', () => {
+      const req = mockRequest({});
+      const result = utils.getAbsoluteUrl(req as ClientRequest);
+      assert.strictEqual(result, 'http://localhost/');
+    });
+
+    it("{ path: '/helloworld', port: 8080 } should return http://localhost:8080/helloworld", () => {
+      const req = mockRequest({
+        path: '/helloworld',
+        port: 8080,
+      });
+      const result = utils.getAbsoluteUrl(req as ClientRequest);
+      assert.strictEqual(result, 'http://localhost:8080/helloworld');
+    });
+
+    it('should return auth credentials as REDACTED to avoid leaking sensitive information', () => {
+      const req = mockRequest({
+        path: '/helloworld',
+        port: 8080,
+        authHeader: 'Basic dXNlcjpwYXNzd29yZA==',
+      });
+      const result = utils.getAbsoluteUrl(req as ClientRequest);
+      assert.strictEqual(
+        result,
+        'http://REDACTED:REDACTED@localhost:8080/helloworld'
+      );
+    });
+
+    it('should return auth credentials and particular query strings as REDACTED', () => {
+      const req = mockRequest({
+        path: '/registers?X-Goog-Signature=secret123',
+        port: 8080,
+        authHeader: 'Basic dXNlcjpwYXNz',
+      });
+      const result = utils.getAbsoluteUrl(req as ClientRequest);
+      assert.strictEqual(
+        result,
+        'http://REDACTED:REDACTED@localhost:8080/registers?X-Goog-Signature=REDACTED'
+      );
+    });
+
+    it('should return particular query strings as REDACTED', () => {
+      const req = mockRequest({
+        path: '/registers?AWSAccessKeyId=secret123',
+        port: 8080,
+      });
+      const result = utils.getAbsoluteUrl(req as ClientRequest);
+      assert.strictEqual(
+        result,
+        'http://localhost:8080/registers?AWSAccessKeyId=REDACTED'
+      );
+    });
+  });
+
+  describe('setSpanWithError()', () => {
+    it('should have error attributes', () => {
+      const errorMessage = 'test error';
+      const error = new Error(errorMessage);
+      const span = {
+        setAttribute: () => undefined,
+        setStatus: () => undefined,
+        recordException: () => undefined,
+      } as unknown as Span;
+      const mock = sinon.mock(span);
```

Comment:
I'd not bring in sinon and use Node.js internal mock module:  https://nodejs.org/api/test.html#mocking